packages/client/components/SwipeableDashSidebar.tsx

```diff
@@ -63,20 +63,28 @@ const updateSpeed = (clientX: number) => {
   swipe.lastX = clientX
 }
 
-const updateIsSwipe = (clientX: number, clientY: number) => {
+const updateIsSwipe = (clientX: number, clientY: number, isRightSidebar: boolean) => {
   const movementX = swipe.startX - clientX
   const movementY = swipe.startY - clientY
   const dx = Math.abs(movementX)
   const dy = Math.abs(movementY)
   const rads = -Math.atan(movementX / dy)
   if (dx > UNCERTAINTY_THRESHOLD || dy > UNCERTAINTY_THRESHOLD) {
-    // if it's open & it's a swipe to the left || it's closed & it's a swipe to the right
-    swipe.isSwipe = swipe.isOpen ? rads <= -MIN_ARC_RADS : rads >= MIN_ARC_RADS
+    const swipingLeft = rads <= -MIN_ARC_RADS
+    const swipingRight = rads >= MIN_ARC_RADS
+    // if it's open & sidebar is on the right & it's a swipe to the right || it's closed
+    // & sidebar is on the right & it's a swipe to the left. Vice versa for left sidebar
+    swipe.isSwipe = swipe.isOpen
```

Comment:
Any preference/thoughts on the nested ternary vs a function / lookup? I don't love the look of it but in this example I find it slightly cleaner & easier to read. I think this guy has [a point](https://medium.com/javascript-scene/nested-ternaries-are-great-361bddd0f340
)

Reply:
darn combinatorics. fwiw i think you wrote it as cleanly as it could be written!